xbmc/settings/SubtitlesSettings.cpp

```diff
@@ -70,3 +104,54 @@ void CSubtitlesSettings::OnSettingChanged(const std::shared_ptr<const CSetting>&
   SetChanged();
   NotifyObservers(ObservableMessageSettingsChanged);
 }
+
+void CSubtitlesSettings::SettingOptionsSubtitleFontsFiller(const SettingConstPtr& setting,
+                                                           std::vector<StringSettingOption>& list,
+                                                           std::string& current,
+                                                           void* data)
+{
+  // Find additionals fonts from the user fonts folder
+  std::vector<std::string> fontCollection;
+  CFileItemList items;
+  XFILE::CDirectory::GetDirectory("special://home/media/Fonts/", items, "",
+                                  XFILE::DIR_FLAG_DEFAULTS);
+  std::mutex vecFontMutex;
+  //!@todo FIXME: this "for" loop and threads in should be replaced with the
+  //! more performant parallel execution std::for_each(std::execution::par, ...
+  //! to halving loading times of fonts list, maybe with C++17 with appropriate
+  //! fix to include parallel execution or with C++20
+  const int itemsSize = items.Size();
+  for (int i = 0; i < itemsSize; ++i)
+  {
+    std::thread t1{&addFont, std::ref(vecFontMutex), items[i], std::ref(fontCollection)};
```

Comment:
I am unsure about this. Someone else should take a look. Is JobManager not viable here?

Reply:
I'm not really sure what our stance is on using `std::thread` directly outside of `CThread`

Reply:
Also, as said on Slack, maybe it's just better to do this once at startup and if new fonts are added then restarting kodi is required.

Reply:
it would make sense to have a some sort of font cache,
i don't think i have enough kodi knowledge to integrate a cache management to be run at startup
also in how/where to store the list of font metadata

Reply:
FontManager would be the place for that I guess

Reply:
removed threads and used JobManager on startup